packages/js/src/executors/verdaccio/verdaccio.impl.ts

```diff
@@ -0,0 +1,188 @@
+      'Verdaccio is not installed. Please run `npm install verdaccio` or `yarn add verdaccio`'
+    );
+  }
+
+  if (options.storage && existsSync(options.storage)) {
+    removeSync(options.storage);
+  }
+  const cleanupFunctions = [setupNpm(options), setupYarn(options)];
+
+  const processExitListener = (signal?: number | NodeJS.Signals) => {
+    if (childProcess) {
+      childProcess.kill(signal);
+    }
+    for (const fn of cleanupFunctions) {
+      fn();
+    }
+  };
+  process.on('exit', processExitListener);
+  process.on('SIGTERM', processExitListener);
+  process.on('SIGINT', processExitListener);
+  process.on('SIGHUP', processExitListener);
+
+  try {
+    await startVerdaccio(options);
+  } catch (e) {
+    logger.error('Failed to start verdaccio: ' + e.toString());
+    return {
+      success: false,
+    };
+  }
+  return {
+    success: true,
+  };
+}
+
+/**
+ * Fork the verdaccio process: https://verdaccio.org/docs/verdaccio-programmatically/#using-fork-from-child_process-module
+ */
+function startVerdaccio(options: VerdaccioExecutorSchema) {
+  return new Promise((resolve, reject) => {
+    childProcess = fork(
+      require.resolve('verdaccio/bin/verdaccio'),
+      createVerdaccioOptions(options),
+      {
+        env: {
+          ...process.env,
+          VERDACCIO_HANDLE_KILL_SIGNALS: 'true',
+          ...(options.storage
+            ? { VERDACCIO_STORAGE_PATH: options.storage }
+            : {}),
+        },
+        stdio: ['inherit', 'pipe', 'pipe', 'ipc'],
+      }
+    );
+
+    childProcess.stdout.on('data', (data) => {
+      process.stdout.write(data);
+    });
+    childProcess.stderr.on('data', (data) => {
+      if (data.includes('VerdaccioWarning')) {
+        process.stdout.write(data);
+      } else {
+        reject(data);
+      }
+    });
+    childProcess.on('error', (err) => {
+      reject(err);
+    });
+    childProcess.on('disconnect', (err) => {
+      reject(err);
+    });
+    childProcess.on('exit', (code) => {
+      if (code === 0) {
+        resolve(code);
+      } else {
+        reject(code);
+      }
+    });
+  });
+}
+
+function createVerdaccioOptions(options: VerdaccioExecutorSchema) {
+  const verdaccioArgs: string[] = [];
+  if (options.port) {
+    verdaccioArgs.push('--listen', options.port.toString());
+  }
+  if (options.config) {
+    verdaccioArgs.push('--config', options.config);
+  }
+  return verdaccioArgs;
+}
+
+function setupNpm(options: VerdaccioExecutorSchema) {
+  try {
+    execSync('npm --version');
+  } catch (e) {
+    return () => {};
+  }
+
+  let npmRegistryPath;
+  try {
+    npmRegistryPath = execSync(
+      `npm config get registry --location ${options.location}`
+    )
+      ?.toString()
+      ?.trim();
+    execSync(
+      `npm config set registry http://localhost:${options.port}/ --location ${options.location}`
+    );
+    execSync(
+      `npm config set //localhost:${options.port}/:_authToken="secretVerdaccioToken"`
+    );
```

Comment:
Can we add this for yarn as well?

Reply:
i read form https://verdaccio.org/docs/setup-yarn/ that yarn just uses `.npmrc` as the default. i think it gets the auth toekn from .npmrc. i don't think there is way to set it in .yarnrc